File: Python/044.py
from Utils import pentagonalNumber, isPentagonalNumber

def find():
    pentagonal = []
    i = 1
    while True:
        pent = pentagonalNumber(i)
        for j in pentagonal:
            if isPentagonalNumber(abs(pent-j)) and isPentagonalNumber(pent+j):
                return (pent-j)
        pentagonal.append(pent)
        i += 1

print(find())

File: C++/044.cpp
#include <iostream>
#include "Utils.h"

int main() {
  for (int i=0; i<100; i++) {
    cout << i << ": " << pentagonalNumber(i)
    << ", " << isPentagonalNumber(pentagonalNumber(i)) << "\n";

  }

}
